lib/v8.js

```diff
@@ -42,3 +51,21 @@ exports.getHeapStatistics = function() {
 };
 
 exports.setFlagsFromString = v8binding.setFlagsFromString;
+
+exports.getHeapSpaceStatistics = function() {
+  const heapSpaceStatistics = [];
+  const buffer = heapSpaceStatisticsBuffer;
+
+  for (let i = 0; i < kNumberOfHeapSpaces; i++) {
+    const heapSpaceName = v8binding.updateHeapSpaceStatisticsArrayBuffer(i);
+    heapSpaceStatistics.push({
+      space_name: heapSpaceName,
+      space_size: buffer[kSpaceSizeIndex],
+      space_used_size: buffer[kSpaceUsedSizeIndex],
+      space_available_size: buffer[kSpaceAvailableSizeIndex],
+      physical_space_size: buffer[kPhysicalSpaceSizeIndex]
+    });
```

Comment:
Similarly, you could allocate `heapSpaceStatistics` with `new Array(kNumberOfHeapSpaces)`, although that's probably a lesser win; IIRC, V8 preallocates 16 elements the first time you push to an empty array.